hadoop-hdfs-project/hadoop-hdfs-httpfs/src/test/java/org/apache/hadoop/lib/util/TestCheck.java

```diff
@@ -116,6 +116,16 @@ public void validIdentifierInvalid3() throws Exception {
     Check.validIdentifier("1", 1, "");
   }
 
+  @Test(expected = IllegalArgumentException.class)
+  public void validIdentifierInvalid4() throws Exception {
+    Check.validIdentifier("`a", 1, "");
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void validIdentifierInvalid5() throws Exception {
+    Check.validIdentifier("[a", 1, "");
+  }
+
```

Comment:
These tests are passing without your fix also

Reply:
Actually I just added the test cases to validate starting characters which should not be allowed as per A-Z but were allowed with A-z

Reply:
The test added should be like: Without your fix it should fail. With your fix it should pass

Reply:
Sure, thanks @ayushtkn. Let me fix the current maxLength and would fix the issue other issue as well next